app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class PostsController < ApplicationController
+  # before_action :set_post, only: [:destroy]
+  before_action :authenticate_user!, except: [:index]
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    @post = current_user.posts.build
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to root_path, notice: 'Post was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # def destroy
+  #   @post.destroy
+  #   respond_to do |format|
+  #     format.html { redirect_to root_path, notice: "Post was successfully deleted." }
+  #   end
+  # end
```

Comment:
It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history